doc/developer/design/20220330_persist.md

```diff
@@ -511,25 +573,24 @@ It's easiest to reason about writing out the entire state of the root of our
 immutable data structure. Sadly, there is an unresolveable tension between
 writes (which append to our list of batches) and compactions (which reduce the
 size of the list of batches). We don't want compaction progress to throttle
-writes, but if writes get ahead of compactions, we get WIP behavior in the
+writes, but if writes get ahead of compactions, we get explosive behavior in the
```

Comment:
quadratic? like, if you write N batches, you have to now write the metadata N times, and end up writing a quadratic amount of metadata bytes compared to how many bytes you would have needed to with incremental metadata?

Reply:
you're correct. done